js_modules/dagster-ui/packages/ui-core/src/partitions/useMatrixData.tsx

```diff
@@ -194,29 +186,47 @@ export type MatrixData = ReturnType<typeof buildMatrixData>;
  *
  * @param inputs
  */
-export const useMatrixData = (inputs: MatrixDataInputs) => {
-  const cachedMatrixData = useRef<{
-    result: MatrixData;
-    inputs: MatrixDataInputs;
-  }>();
-  if (!inputs.solidHandles) {
-    return null;
-  }
-  if (cachedMatrixData.current && shallowCompareKeys(inputs, cachedMatrixData.current.inputs)) {
-    return cachedMatrixData.current.result;
-  }
+export const useMatrixData = ({
+  solidHandles,
+  stepQuery,
+  partitionNames,
+  partitions,
+  options,
+}: MatrixDataInputs) => {
+  const nodes = useMemo(() => {
+    if (!solidHandles) {
+      return [];
+    }
+    return explodeCompositesInHandleGraph(solidHandles).map((h) => h.solid);
+  }, [solidHandles]);
 
-  const nodes = explodeCompositesInHandleGraph(inputs.solidHandles).map((h) => h.solid);
+  const solidsFiltered = useMemo(() => {
+    // Filter the pipeline's structure and build the flat gantt layout for the left hand side
+    return filterByQuery(nodes, stepQuery);
+  }, [nodes, stepQuery]);
 
-  // Filter the pipeline's structure and build the flat gantt layout for the left hand side
-  const solidsFiltered = filterByQuery(nodes, inputs.stepQuery);
+  const layout = useMemo(() => {
+    return buildLayout({nodes: solidsFiltered.all, mode: GanttChartMode.FLAT});
+  }, [solidsFiltered.all]);
 
-  const layout = buildLayout({nodes: solidsFiltered.all, mode: GanttChartMode.FLAT});
+  const partitionsByName = useMemo(() => {
+    const partitionsByName: Record<string, PartitionRuns> = {};
+    partitions.forEach((p) => {
+      // sort partition runs in place
+      p.runs.sort(byStartTimeAsc);
+      partitionsByName[p.name] = p;
+    });
+    return partitionsByName;
+  }, [partitions]);
 
-  // Build the matrix of step + partition squares - presorted to match the gantt layout
-  const result = buildMatrixData(layout, inputs.partitionNames, inputs.partitions, inputs.options);
-  cachedMatrixData.current = {result, inputs};
-  return result;
+  return useThrottledMemo(
```

Comment:
I think the throttling here makes a ton of sense BUT this hook only returns the result of the throttled memo, which means that computing the stuff above on every render doesn't buy us anything -- maybe we move the useThrottledMemo so that it's around the entirety of useMatrixData?

Reply:
The stuff above is very cheap to compute but yeah I can just throttle the whole thing